File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/java/SumOfPrimes.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;


class SumOfPrimes {
    
    /*@
    public normal_behavior
      ensures \result == (k >= 2 && (\forall int d; 2 <= d && (\bigint)d * d <= k; k % d != 0));
    model public static pure boolean fb_prime(int k);
    public normal_behavior
      ensures \result == (n < 2 ? 0 : fb_primeSum(n - 1) + (fb_prime(n) ? n : 0));
      measured_by n < 2 ? 0 : n;
    model public static pure \bigint fb_primeSum(int n);
    public normal_behavior
      ensures \result == (\forall int d; 2 <= d < limit && (\bigint)d * d <= k; k % d != 0);
    model public static pure boolean fb_unmarked(int k, int limit);

    public normal_behavior
      requires -1 <= n <= 1;
      assignable \nothing;
      ensures \result == 0;
      also
    public normal_behavior
      requires 2 <= n < 46349 && System.out != null;
      assignable System.out.outputText, System.out.endsInNewLine;
      ensures \result == fb_primeSum(n);
      ensures System.out.outputText.startsWith(\old(System.out.outputText));
      ensures System.out.endsInNewLine;
      also
    public exceptional_behavior
      requires n < -1 || n == Integer.MAX_VALUE;
      assignable \nothing;
      signals_only NegativeArraySizeException;
      also
    public exceptional_behavior
      requires 2 <= n < Integer.MAX_VALUE && System.out == null;
      assignable \nothing;
      signals_only NullPointerException;
      also
    public exceptional_behavior
      requires 46349 <= n < Integer.MAX_VALUE && System.out != null;
      assignable System.out.outputText, System.out.endsInNewLine;
      signals_only ArrayIndexOutOfBoundsException;
    @*/
    //@ code_java_math
    public static int sumOfPrimes(int n) {
        int sum = 0;
        boolean[] isPrime = new boolean[n + 1];
        Arrays.fill(isPrime, true);
        /*@
        loop_invariant 2 <= i <= 46349 && (\bigint)i <= (n < 2 ? 2 : (\bigint)n + 1);
        loop_invariant sum == fb_primeSum(i - 1);
        loop_invariant (\forall int k; 2 <= k <= n; isPrime[k] == fb_unmarked(k, i));
        loop_writes i, sum, isPrime[*], System.out.outputText, System.out.endsInNewLine;
        decreases n < 2 ? 0 : (\bigint)n - i + 1;
        @*/
        for (int i = 2; i <= n; i++) {
            if (isPrime[i]) {
                //@ assert fb_prime(i);
                System.out.println(i);
                sum += i;
                /*@
                loop_invariant n < 46349 ==> (\bigint)i * i <= j && (\bigint)j <= (n < (\bigint)i * i ? (\bigint)i * i : (\bigint)n + i);
                loop_invariant n < 46349 ==> j % i == 0;
                loop_invariant n < 46349 ==> (\forall int k; 2 <= k <= n; isPrime[k] == (fb_unmarked(k, i) && !((\bigint)i * i <= k < j && k % i == 0)));
                loop_writes j, isPrime[*];
                decreases 4294967296L - (j < 0 ? (\bigint)j + 4294967296L : (\bigint)j);
                @*/
                for (int j = i * i; j <= n; j += i) {
                    isPrime[j] = false;
                }
            }
        }
        return sum;
    }
}

File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/c/SumOfPrimes.c
#include "java_arrays.h"
#include <inttypes.h>
#include <stdint.h>
#include <stdio.h>

/*@
assigns \nothing;
ensures \result == (int32_t)((integer)left + right);
*/
static int32_t java_add(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left + (uint32_t)right);
}

/*@
assigns \nothing;
ensures \result == (int32_t)((integer)left * right);
*/
static int32_t java_mul(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left * (uint32_t)right);
}

/*@
predicate fb_prime(integer k) = k >= 2 && (\forall integer d; 2 <= d && d * d <= k ==> k % d != 0);
logic integer fb_prime_sum(integer n) = n < 2 ? 0 : fb_prime_sum(n - 1) + (fb_prime(n) ? n : 0);
predicate fb_unmarked(integer k, integer limit) = \forall integer d; 2 <= d < limit && d * d <= k ==> k % d != 0;
*/
/*@
requires -1 <= n < 46349;
ensures \result == fb_prime_sum(n);
behavior silent:
  assumes n < 2;
  assigns \nothing;
behavior printing:
  assumes n >= 2;
  assigns *__fc_stdout;
complete behaviors;
disjoint behaviors;
*/
int32_t sumOfPrimes(int32_t n) {
    int32_t length = java_add(n, 1);
    JBoolArray isPrime = jbool_array_new(length);
    int32_t arrayLength = jbool_array_length(isPrime);
    /*@
    loop invariant 0 <= i <= arrayLength;
    loop invariant jboolarray_valid_nonnull(isPrime) && isPrime->length == arrayLength && arrayLength == n + 1;
    loop invariant \forall integer k; 0 <= k < i ==> isPrime->data[k];
    loop assigns i, isPrime->data[0 .. arrayLength - 1];
    loop variant arrayLength - i;
    */
    for (int32_t i = 0; i < arrayLength; i = java_add(i, 1)) {
        jbool_array_set(isPrime, i, true);
    }

    int32_t sum = 0;
    /*@
    loop invariant 2 <= i <= \max(2, n + 1);
    loop invariant jboolarray_valid_nonnull(isPrime) && isPrime->length == n + 1;
    loop invariant sum == fb_prime_sum(i - 1);
    loop invariant \forall integer k; 2 <= k <= n ==> (isPrime->data[k] <==> fb_unmarked(k, i));
    loop assigns i, sum, isPrime->data[0 .. n], *__fc_stdout;
    loop variant \max(0, n - i + 1);
    */
    for (int32_t i = 2; i <= n; i = java_add(i, 1)) {
        if (jbool_array_get(isPrime, i)) {
            /*@ assert fb_prime(i); */
            printf("%" PRId32 "\n", i);
            sum = java_add(sum, i);
            /*@
            loop invariant (integer)i * i <= j <= \max((integer)i * i, (integer)n + i);
            loop invariant j % i == 0;
            loop invariant jboolarray_valid_nonnull(isPrime) && isPrime->length == n + 1;
            loop invariant \forall integer k; 2 <= k <= n ==> (isPrime->data[k] <==> (fb_unmarked(k, i) && !((integer)i * i <= k < j && k % i == 0)));
            loop assigns j, isPrime->data[0 .. n];
            loop variant \max(0, n - j + 1);
            */
            for (int32_t j = java_mul(i, i); j <= n; j = java_add(j, i)) {
                jbool_array_set(isPrime, j, false);
            }
        }
    }
    return sum;
}
